src/arrayvec.rs

```diff
@@ -648,6 +648,72 @@ impl<A: Array> ArrayVec<A> {
     new
   }
 
+  /// Creates a splicing iterator that removes the specified range in the
+  /// vector, yields the removed items, and replaces them with elements from
+  /// the provided iterator.
+  ///
+  /// ## Panics
+  /// * If the start is greater than the end.
+  /// * If the end is past the edge of the vec.
+  /// * If the provided iterator panics.
+  /// * If the new length would overflow the capacity of the array. Because
+  ///   `ArrayVecSplice` adds elements to this vec in its destructor when
+  ///   necessary, this panic would occur when it is dropped.
+  ///
+  /// ## Example
+  /// ```rust
+  /// use tinyvec::*;
+  /// let mut av = array_vec!([i32; 4] => 1, 2, 3);
+  /// let av2: ArrayVec<[i32; 4]> = av.splice(1.., 4..=6).collect();
+  /// assert_eq!(av.as_slice(), &[1, 4, 5, 6][..]);
+  /// assert_eq!(av2.as_slice(), &[2, 3][..]);
+  ///
+  /// av.splice(.., None);
+  /// assert_eq!(av.as_slice(), &[]);
+  /// ```
+  #[inline]
+  pub fn splice<R, I>(
+    &mut self,
+    range: R,
+    replacement: I,
+  ) -> ArrayVecSplice<'_, A, I::IntoIter>
+  where
+    R: RangeBounds<usize>,
+    I: IntoIterator<Item = A::Item>,
+    I::IntoIter: FusedIterator,
```

Comment:
The [splice](https://doc.rust-lang.org/alloc/vec/struct.Vec.html#method.splice) in Vec doesn't have this extra FusedIterator bound. Is it just because the iterator might overflow the vec?

Reply:
I just added it for clarity, I really think `Vec::splice` should have the same bound. If it's not fused, anything after the last `None` will be ignored anyway.

If you want, I can get rid of it.

Reply:
You are correct that the normal splice should have it, but I'm worried about people getting very confusing errors.

How about take it out but put a reminder in the method's comments that anything after the first `None` will be ignored.

Reply:
Will do